Sources/SwiftKuery/Connection.swift

```diff
@@ -68,24 +75,22 @@ public protocol Connection {
     
     /// Execute a raw query with parameters.
     ///
-    /// - Parameter query: A String with the query to execute.
+    /// - Parameter raw: A String with the query to execute.
     /// - Parameter parameters: A dictionary of the parameters with parameter names as the keys.
     /// - Parameter onCompletion: The function to be called when the execution of the query has completed.
     func execute(_ raw: String, parameters: [String:Any?], onCompletion: @escaping ((QueryResult) -> ()))
     
     /// Prepare statement.
     ///
     /// - Parameter query: The query to prepare statement for.
-    /// - Returns: The prepared statement.
-    /// - Throws: QueryError.syntaxError if query build fails, or a database error if it fails to prepare statement.
-    func prepareStatement(_ query: Query) throws -> PreparedStatement
+    /// - Parameter onCompletion: The function to be called when the statement has been prepared.
+    func prepareStatement(_ query: Query, onCompletion: @escaping ((PreparedStatement?, QueryError?) -> ()))
 
     /// Prepare statement.
     ///
     /// - Parameter raw: A String with the query to prepare statement for.
-    /// - Returns: The prepared statement.
-    /// - Throws: QueryError.syntaxError if query build fails, or a database error if it fails to prepare statement.
-    func prepareStatement(_ raw: String) throws -> PreparedStatement
+    /// - Parameter onCompletion: The function to be called when the statement has been prepared.
+    func prepareStatement(_ raw: String, onCompletion: @escaping ((PreparedStatement?, QueryError?) -> ()))
```

Comment:
Make this return a QueryResult with success embedding preparedStatement, add helper to retrieve statement called asPreparedStatement.